pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -692,29 +692,24 @@ func (aws *AWSCloud) NodeAddresses(name string) ([]api.NodeAddress, error) {
 }
 
 // ExternalID returns the cloud provider ID of the specified instance (deprecated).
-// Note that if the instance does not exist or is no longer running, we must return ("", cloudprovider.InstanceNotFound)
+// We assume that if this is run on the instance itself, the instance exists and is alive
 func (aws *AWSCloud) ExternalID(name string) (string, error) {
-	// We must verify that the instance still exists
-	instance, err := aws.findInstanceByNodeName(name)
+	awsInstance, err := aws.getSelfAWSInstance()
```

Comment:
I don't think you can assume we're requesting the local instance.  Instead, you need to get the local instance with getSelfAWSInstance, check if the name matches, and if so return immediately.  Otherwise proceed with the method as it used to be..